go/worker/compute/merge/committee/node.go

```diff
@@ -273,8 +280,28 @@ func (n *Node) HandleNewBlockLocked(blk *block.Block) {
 // HandleResultsFromExecutorWorkerLocked processes results from an executor worker.
 // Guarded by n.commonNode.CrossNode.
 func (n *Node) HandleResultsFromExecutorWorkerLocked(spanCtx opentracing.SpanContext, commit *commitment.ExecutorCommitment) {
-	// TODO: Context.
-	if err := n.handleResultsLocked(context.TODO(), commit); err != nil {
+	ctx, err := n.commonNode.Group.RoundContext()
+	if err != nil {
+		// RoundContext can return an error if there is no active epoch,
+		// which shouldn't happen in our case.
+		n.logger.Warn("HandleResultsFromExecutorWorkerLocked called with no active epoch")
+		return
+	}
+
+	call := func() error {
+		grr := n.handleResultsLocked(*ctx, commit)
+		switch grr {
+		case errIncorrectState:
+			return backoff.Permanent(grr)
```

Comment:
Why no retry in this case? The P2P code path retries even if the state is incorrect.

Reply:
Fixed :)